src/QtAVPlayer/qavdemuxer.cpp

```diff
@@ -109,6 +109,45 @@ class QAVDemuxerPrivate
     QString bsfs;
 };
 
+static void log_callback(void *ptr, int level, const char *fmt, va_list vl)
+{
+    /* Do we need to log ? */
+    if(level > av_log_get_level()){
+        return;
+    }
+
+    /* Format log line */
+    char line[1024];
+    static int print_prefix = 1;
+
+    av_log_format_line(ptr, level, fmt, vl, line, sizeof(line), &print_prefix);
+
+    /* Adapt it to Qt log format */
+    switch(level)
+    {
+        case AV_LOG_PANIC:
+        case AV_LOG_FATAL:{
+            qFatal("[ffmpeg] %s", line);
```

Comment:
For `AV_LOG_FATAL`, we could use qCritical, you are right. But for `AV_LOG_PANIC`, I think we should stick with `qFatal` since [FFmpeg will still crash](https://ffmpeg.org/doxygen/trunk/group__lavu__log__constants.html#ga09623098406ba81d130824ac742d146e)

Reply:
```
QFATAL : tst_QAVPlayer::multipleVideoStreams(7_BCL02006_ffv1_20s_2.mkv) [ffmpeg] [mjpeg @ 0x7fade800f0c0] No JPEG data found in image

```
https://github.com/valbok/QtAVPlayer/actions/runs/12158943545/job/33909140955?pr=499